src/IO/ADIOS/ADIOS2IOHandler.cpp

```diff
@@ -1209,10 +1209,13 @@ ADIOS2IOHandlerImpl::verifyDataset( Offset const & offset,
     {
         auto requiredType = adios2::GetType< T >( );
         auto actualType = IO.VariableType( varName );
-        VERIFY_ALWAYS( requiredType == actualType,
-                       "[ADIOS2] Trying to access a dataset with wrong type (trying to "
-                       "access dataset with type " +
-                           requiredType + ", but has type " + actualType + ")" )
+        std::stringstream errorMessage;
+        errorMessage
+            << "[ADIOS2] Trying to access a dataset with wrong type (trying to "
+               "access dataset with type "
+            << determineDatatype< T >() << ", but has type "
+            << detail::fromADIOS2Type( actualType ) << ")";
```

Comment:
Did you mean to add `false` here?
```suggestion
            << detail::fromADIOS2Type( actualType, false ) << ")";
```

Reply:
Ah yes, forgot that, thank you!